truffle/src/main/java/org/jruby/truffle/translator/BodyTranslator.java

```diff
@@ -415,6 +415,13 @@ public RubyNode visitCallNode(CallNode node) {
                 && ((org.jruby.ast.Colon2ConstNode) node.getReceiverNode()).getName().equals("Primitive")
                 && node.getName().equals("assert_not_compiled")) {
             return AssertNotCompiledNodeGen.create(context, sourceSection);
+        } else if (node.getReceiverNode() instanceof org.jruby.ast.ConstNode
+                && ((ConstNode) node.getReceiverNode()).getName().equals("Truffle")) {
+            if (node.getName().equals("omit")) {
+                // We're never going to run the omitted code and it's never used as the RHS for anything, so just
+                // replace the call with nil.
+                return new LiteralNode(context, sourceSection, context.getCoreLibrary().getNilObject());
+            }
```

Comment:
We do need to extract some local vars in the Translator, it becomes really hard to read.